firmware/Core/Src/config/configuration.c

```diff
@@ -0,0 +1,137 @@
+#include "config/configuration.h"
+#include "eps/eps_config.h"
+
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+#include "configuration.h"
+
+uint32_t thing1 = 13345;
+uint32_t thing2 = 242344;
+// extern
+CONFIG_integer_config_entry_t CTS1_Integer_Configuration_Variables[] = {
+    {
+     .variable_name = "test_integer_config_eps",
+     .num = &test_integer_config_eps,
+     .width_bytes = sizeof(test_integer_config_eps)
+    },
+    {
+     .variable_name = "thing1",
+     .num = &thing1,
+     .width_bytes = sizeof(thing1)
+    },
+    {
+     .variable_name = "thing2",
+     .num = &thing2,
+     .width_bytes = sizeof(thing2)
+    }
+
+};
+
+// extern
+const uint8_t CTS1_Integer_Configuration_Variables_Count = sizeof(CTS1_Integer_Configuration_Variables) / sizeof(CONFIG_integer_config_entry_t);
+
+// extern
+CONFIG_string_config_entry_t CTS1_String_Configuration_Variables[] = {
+    {
+
+        .max_length = 10,
+        .variable_name = "test_string_config_eps",
+        .variable_pointer = test_string_config_eps
+
+    }
+
+};
+
+// extern
+const uint8_t CTS1_String_Configuration_Variables_Count = sizeof(CTS1_String_Configuration_Variables) / sizeof(CONFIG_string_config_entry_t);
+
+/// @brief Finds an integer configuration variable by name
+/// @param name Name of the variable
```

Comment:
Missing params in docstring. Why is `result` a double pointer?  We probably don't need `str_len` if `name` is agreed to be a cstring.